### ISSUE TITLE
linter: only supports 1x `tsconfig.json` causing limited support for import plugin in monorepos

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#12492]
The import plugin relies on oxc-resolver, which relies on the tsconfig.json for correct module resolution. If a `tsconfig.json` is not supplied, the module resolution may not be accurate.
Currently we only support 1x tsconfig json (that must - to confirm be supplied by the cli args) to customize this.

In future (especially with the consideration of type aware linting), i think we should auto discover tsconfigs and use the closest to the file (if reasonable - how does typescript's LSP work when decifing which tsconfig to use?_

Original Ref:

> oxlint not the lang server supports nested tsconfigs at present. It's something i brought up in relation to type aware linting.
> WIll make an issue to track support 

 _Originally posted by @camc314 in [#12449](https://github.com/oxc-project/oxc/issues/12449#issuecomment-3112589334)_

**Comment by overlookmotel** [comment]
I'm not sure, but I wonder if this would ideally be implemented in `oxc-resolver`? Otherwise, if I understand right, we'll need to have a separate `Resolver`s for each `tsconfig.json`. Each `Resolver` will have a separate cache, which would result in duplicated work and higher memory usage.

**Comment by camc314** [comment]
https://github.com/oxc-project/oxc/pull/12552#issuecomment-3125604559

Boshen had the same comment.

I didn't really think about that when i was working on this, but now, thinking about it, it makes sense to occur in oxc_resolver.

Doing it in oxc_resolver should also allow us to remove this horrible code:


```rs
    /// Cache of resolvers per tsconfig path
    resolver_cache: Arc<Mutex<FxHashMap<Option<PathBuf>, Arc<Resolver>>>>,
    /// Cache of tsconfig paths for directories
    tsconfig_cache: Arc<Mutex<FxHashMap<PathBuf, Option<PathBuf>>>>,
```

**Comment by camc314** [comment]
Relevant TS code:

https://github.com/microsoft/typescript/blob/b8e4ed8aeb0b228f544c5736908c31f136a9f7e3/src/server/editorServices.ts#L2579-L2631

https://github.com/microsoft/typescript/blob/b8e4ed8aeb0b228f544c5736908c31f136a9f7e3/src/server/editorServices.ts#L2538

**Comment by ChangeHow** [comment]
Any updates for current feature? type-aware is awesome, it help found more ts lint issues, but cycle deps seems strong depend on this feature?

**Comment by Boshen** [comment]
Fixed by https://github.com/oxc-project/oxc/pull/17489
